--- pdfreader.py.orig	2016-09-05 20:33:48.626874677 -0500
+++ pdfreader.py	2016-09-05 20:33:59.759081856 -0500
@@ -232,7 +232,7 @@
         # just return it if it is a simple object.
         obj.indirect = key
         tok = source.next()
-        if tok == 'endobj':
+        if tok == 'endobj' and not tok.isdigit():
             return obj
 
         # Should be a stream.  Either that or it's broken.
