Index: rst2pdf/createpdf.py
===================================================================
--- rst2pdf/createpdf.py	(revisión: 2632)
+++ rst2pdf/createpdf.py	(copia de trabajo)
@@ -74,7 +74,7 @@
 
 from reportlab.platypus import *
 from reportlab.platypus.doctemplate import IndexingFlowable
-from reportlab.platypus.flowables import _listWrapOn, _Container
+from reportlab.platypus.flowables import _listWrapOn, _Container, KeepTogether
 from reportlab.pdfbase.pdfdoc import PDFPageLabel
 #from reportlab.lib.enums import *
 #from reportlab.lib.units import *
@@ -646,6 +646,24 @@
             if not isinstance(elements[-1],UnhappyOnce):
                 log.info ('Forcing second pass so Total pages work')
                 elements.append(UnhappyOnce())
+
+        # fix widowes here
+        new_story = []
+        elements = iter(elements)
+        for e in elements:
+            if getattr(e, "level", 0) > 0:
+                # a title! bind together it and the next(s) element(s) (until
+                # we get a non-title)
+                together = [e]
+                for e in elements:
+                    together.append(e)
+                    if getattr(e, "level", 0) == 0:
+                        break
+                new_story.append(KeepTogether(together))
+            else:
+                new_story.append(e)
+        elements = new_story
+
         while True:
             try:
                 log.info("Starting build")
@@ -678,7 +696,10 @@
                             if fnPile:
                                 fnPile.insert(0, Separation())
                                 newStory.append(Sinker(fnPile))
-                            newStory.append(e)
+                                if not isinstance(e, MyPageBreak):
+                                    newStory.append(e)
+                            else:
+                                newStory.append(e)
                             fnPile=[]
                         else:
                             newStory.append(e)
